### PR TITLE
feat: build with deno; improve theme coverage

### LINKED ISSUE #4: Theming the "Spaces Toolbar"
By default, the themes here do not theme the "Spaces Toolbar" (pictured below, themed manually).

By right clicking => customise, the user can set the relevant colours.
If this cannot be themed by an xpi, this process should probably be documented in the README. Otherwise, it would be good to set it up automatically!

image

### LINKED ISSUE #5: Unique ids
The ID's for all themes in this repository are the same value, namely: `elkrien@gmail.com`. Installing multiple of these themes (at least from the cli) is therefore impossible as far as I know.

Would it be possible to make the id's unique so that we can install multiple?

### LINKED ISSUE #7: Several things arent themed
This includes for example the new message button which stays the normal blue or the right click menu

### PR DESCRIPTION
**Pull request by nullishamy**
This brings the build system up to date with current Catppuccin tech, improves the coverage of the theme, and overall improves the quality of things.

Closes #5
Closes #4 
Closes #7

### CODE REVIEW

File: build.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  Object.entries(variants).map(async ([name, palette]) => {
+    for (const accent of accents) {
+      const newTheme = {
+        manifest_version: 2,
+        name: `catppuccin-${name}-${accent}`,
+        version: "1.0.0",
+        applications: {
+          gecko: {
+            // FIXME: Make properly
+            id: `theme@catppuccinorg.com`,
```
Comment (uncenter) [inline build.ts:36]:
```suggestion
            id: `thunderbird@catppuccin.com`,
```

File: build.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            strict_min_version: "60.0",
+          },
+        },
+        description: `Catppuccin theme for Thunderbird - ${titleCase(
```
Comment (uncenter) [inline build.ts:40]:
```suggestion
        description: `Soothing pastel theme for Thunderbird - ${titleCase(
```
Not sure if that sort of description format would work here? :p

File: build.ts (hunk 1 above)
Comment (sgoudham) [reply build.ts:36]: Would this not be an actual email address that we have? E.g.  `releases@catppuccin.com`

Comment (uncenter) [reply build.ts:36]: I think it doesn't have to be an actual email lol.

Comment (nullishamy) [reply build.ts:36]: Thunderbird enforces some scheme similar to email through a regex. Haven't quite thought about what this should actually be yet. Worth noting that #5 should be solved here.

Comment (sgoudham) [reply build.ts]: From Thunderbird's [A Guide To Theming](https://developer.thunderbird.net/add-ons/web-extension-themes):

image

File: flake.nix [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    in 
+      pkgs.mkShell {
+        packages = with pkgs; [
+          just
+          deno
+          catppuccin-toolbox.packages.${system}.whiskers
+        ];
+      };
+  });
+}
```
Comment (sgoudham) [inline flake.nix:21]: Do we need `just` or `whiskers` here?

```suggestion
{
  inputs = {
    nixpkgs.url = "github:nixos/nixpkgs/nixos-23.11";
    flake-utils.url = "github:numtide/flake-utils";
  };

  description = "Development shell flake";
  outputs = { self, nixpkgs, flake-utils }: flake-utils.lib.eachDefaultSystem (system: {
    devShell = let 
      pkgs = nixpkgs.legacyPackages.${system};
    in 
      pkgs.mkShell {
        packages = with pkgs; [
          deno
        ];
      };
  });
}
```

File: build.ts [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+  AlphaColor,
+  Color,
+  Labels,
+  variants,
+} from "npm:@catppuccin/palette@0.1.5";
+import { titleCase } from "https://deno.land/x/case@2.2.0/mod.ts";
+import { JSZip } from "https://deno.land/x/jszip@0.11.0/mod.ts";
+import * as uuid from "https://deno.land/std@0.207.0/uuid/mod.ts";
+
+// DON'T CHANGE THIS :OMEGALUL:
```
Comment (sgoudham) [inline build.ts:12]: Sorry, meant to change this comment before pushing yesterday.

```suggestion
// Allows for the UUIDs to be "seeded" and therefore reproducible
// DO NOT CHANGE
```

File: build.ts [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+    const zip = new JSZip();
+    zip.addFile("manifest.json", json);
+
+    const images = zip.folder("images")
+    images.addFile("icon16.png", "./assets/icon16.png")
+    images.addFile("icon48.png", "./assets/icon48.png")
+    images.addFile("icon128.png", "./assets/icon128.png")
+
+    await zip.writeZip(`./themes/${name}/${name}-${accent}.xpi`);
```
Comment (sgoudham) [inline build.ts:136]: Are we sure that we want the directory structure to be like `mocha/mocha-blue`? I'm not entirely fussed but just looked odd at a glance.

**Review by sgoudham — CHANGES_REQUESTED** [review]

**Review by uncenter — APPROVED** [review]

File: build.ts (hunk 5 above)
Comment (uncenter) [reply build.ts:136]: Ooh right :eyes:

Comment (nullishamy) [reply build.ts:137]: I prefer the names being more descriptive, and we will more than likely condense the entire port into 1 XPI in the future, as discussed on Discord. I think for the time being this is fine?

**Review by sgoudham — APPROVED** [review]

**Pull request merged**